textile/chat-features.textile

```diff
@@ -9,5 +9,148 @@ jump_to:
+*** @(CHA-RS2b1)@ @[Testable]@ If an error is associated with the current status, then this must be exposed.
+*** @(CHA-RS2b2)@ @[Testable]@ If there is no error is associated with the current status, then the room status should not expose any errors.
+* @(CHA-RS3)@ @[Testable]@ A newly created room must have a current status of @INITIALIZED@.
+* @(CHA-RS4)@ A room must allow changes to room status to be observed by clients.
+** @(CHA-RS4a)@ @[Testable]@ Room status update events must contain the newly entered room status.
+** @(CHA-RS4b)@ @[Testable]@ Room status update events must contain the previous room status.
+** @(CHA-RS4c)@ @[Testable]@ Room status update events must not contain an error, where that error pertains to the current status.
+** @(CHA-RS4d)@ @[Testable]@ Room status update events must contain an error, where that error pertains to the current status.
+** @(CHA-RS4e)@ @[Testable]@ Clients must be able to register a listener for room status updates and receive such events.
+** @(CHA-RS4f)@ @[Testable]@ Clients must be able to unregster a listener for room status updates and from that point onwards, cease to receive such events on that listener only.
+** @(CHA-RS4g)@ @[Testable]@ Clients must be able to unregster all listeners for room status updates and from that point onwards, cease to receive such events on all listeners.
+
+h3(#rooms-lifecycle). Room Lifecycle
+
+Rooms are considered the atomic unit of chat and comprise of potentially multiple underlying realtime channels. The status of the room is, broadly speaking, the combination of the respective underlying channel states (the rooms @contributors@). We present one unified status, rather than statuses for multiple individual features, as this is easier for developers to reason about.
+
+Chat features are spread across multiple Realtime Channels. For the purpose of this section, a @contributor@ is a feature in Chat that is considered as part of the overall room status, as well as its lifecycle. A @contributor@ might share its realtime channel with another (e.g. messages and occupancy), or it might be entirely standalone (room reactions). In that sense, implementations of Room Lifecycle @MUST@ make no assumptions over which realtime channels are in use or being shared, and treat every @contributor@ as being entirely standalone.
+
+There are four room lifecycle operations: @ATTACH@, @DETACH@, @RELEASE@ and @RETRY@. @RETRY@ is an internal-only operation. In order to ensure the integrity of the chat room status, these operations are carried out atomically.
+
+h4(#rooms-lifecycle-operations). Room Lifecycle Operations
+
+* @(CHA-RL1)@ A room must be explicitly @ATTACHED@ to.
+** @(CHA-RL1a)@ @[Testable]@ If the room is already in the @ATTACHED@ status, then this operation is no-op.
+** @(CHA-RL1b)@ @[Testable]@ If the room is in the @RELEASING@ status, the operation shall be rejected with error code @102102@.
```

Comment:
It's not clear what "rejected with error code @102102@" means here. (I know from looking at the JS interface that it reuses the `ErrorInfo` type from the core SDK, but it would be good to make that explicit.)

Reply:
Also, if the intention is to reuse `ErrorInfo`, then all of the places where this spec say to throw such an error should also specify the `statusCode`.

Reply:
I can't see where the status codes are specified; it seems like you've only added one for CHA-RL1h2. It seems like there should be status codes specified for:

- everything in the "Chat-specific Error Codes" section
- all the places where we specify a numerical error code directly in the spec point (e.g. CHA-RC1c, CHA-RC2b).

Reply:
I've provided the status codes for chat-specific errors. For generic codes e.g. `40000` - the standard convention of matching the status code to the error code should be followed (in this case, 400).